libbeat/publisher/client.go

```diff
@@ -78,32 +79,49 @@ func newClient(pub *PublisherType) *client {
 }
 
 func (c *client) PublishEvent(event common.MapStr, opts ...ClientOption) bool {
-	c.annotateEvent(event)
-
+	publishEvent := c.alterEvent(event)
+	if publishEvent == nil {
+		return false
```

Comment:
should return true to indicate success. Beat only cares about sending, but should not be affected filtering. Also signal success to Signal callback in ctx.